mautic/mautic
# Hotfix for the Lead Query

Please answer the following questions:

| Q             | A
| ------------- | ---
| Bug fix?      | N
| New feature?  | N
| BC breaks?    | N
| Deprecations? | N
| Fixed issues  |  N/A

## Description

In a leads table with 200K+ records, loading the contacts page can be pretty slow. I discovered this is due to the way we query for anonymous leads. This PR updates the query to check the `date_identified` field instead of checking `email, firstname, lastname, company` fields, which in testing increased query speed by ~3 seconds, which is huge.

## Steps to reproduce and test

Have a big leads table and run in dev mode with the symfony profiler. Check the query tab and see the 4th & 5th query. Note the time then apply this patch and do it again. The times should be significantly lower.

#### Previous
![screen shot 2016-05-19 at 4 30 55 pm](https://cloud.githubusercontent.com/assets/718028/15409646/b208ff98-1de3-11e6-8812-a7060a2ec78c.png)

#### New
![screen shot 2016-05-19 at 4 50 02 pm](https://cloud.githubusercontent.com/assets/718028/15409645/b2080b60-1de3-11e6-8096-43477634761e.png)
